sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/execute/executor/reader/AbstractJdbcReader.java

```diff
@@ -48,46 +48,44 @@
  */
 @Slf4j
 public abstract class AbstractJdbcReader extends AbstractSyncRunner implements JdbcReader {
-
+    
     @Getter(AccessLevel.PROTECTED)
     private final RdbmsConfiguration rdbmsConfiguration;
     
     private final DataSourceFactory dataSourceFactory;
-
+    
     @Setter
     private Channel channel;
-
+    
     public AbstractJdbcReader(final RdbmsConfiguration rdbmsConfiguration, final DataSourceFactory dataSourceFactory) {
         if (!JdbcDataSourceConfiguration.class.equals(rdbmsConfiguration.getDataSourceConfiguration().getClass())) {
             throw new UnsupportedOperationException("AbstractJdbcReader only support JdbcDataSourceConfiguration");
         }
         this.rdbmsConfiguration = rdbmsConfiguration;
         this.dataSourceFactory = dataSourceFactory;
     }
-
+    
     @Override
     public final void run() {
         start();
         read(channel);
     }
-
+    
     @Override
     public final void read(final Channel channel) {
         try (Connection conn = dataSourceFactory.getDataSource(rdbmsConfiguration.getDataSourceConfiguration()).getConnection()) {
             String sql = String.format("select * from %s %s", rdbmsConfiguration.getTableName(), rdbmsConfiguration.getWhereCondition());
             PreparedStatement ps = conn.prepareStatement(sql, java.sql.ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
-            ps.setFetchSize(100);
-            ps.setFetchDirection(ResultSet.FETCH_REVERSE);
+            ps.setFetchSize(Integer.MIN_VALUE);
             ResultSet rs = ps.executeQuery();
             ResultSetMetaData metaData = rs.getMetaData();
             while (isRunning() && rs.next()) {
                 DataRecord record = new DataRecord(new NopLogPosition(), metaData.getColumnCount());
                 record.setType("bootstrap-insert");
                 record.setTableName(rdbmsConfiguration.getTableNameMap().get(rdbmsConfiguration.getTableName()));
                 for (int i = 1; i <= metaData.getColumnCount(); i++) {
-                    if (Types.TIME == rs.getMetaData().getColumnType(i)
-                            || Types.DATE == rs.getMetaData().getColumnType(i)
-                            || Types.TIMESTAMP == rs.getMetaData().getColumnType(i)) {
+                    if ("MySQL".equals(rdbmsConfiguration.getDataSourceConfiguration().getDatabaseType().getName())
+                            && isDateTimeValue(rs.getMetaData().getColumnType(i))) {
```

Comment:
Abstract class should not perceive the specific type, it can use abstract method replace, and let subClass to implement.